pkg/agent/controller/agent.go

```diff
@@ -381,6 +391,13 @@ func (a *Controller) serviceToRemoteServiceImport(obj runtime.Object, numRequeue
 	serviceImport := a.newServiceImport(svcExport.Name, svcExport.Namespace)
 
 	// Update the status and requeue
+	if op == syncer.Update {
+		a.updateExportedServiceStatus(svcExport.Name, svcExport.Namespace, corev1.ConditionFalse, serviceUpdate,
+			"Service update")
```

Comment:
Rather than changing the status condition to an intermediate state indicating it's not valid and relying on a `ServiceExport` update to trigger `serviceExportToServiceImport`, I think it would be cleaner to call `serviceExportToServiceImport`  directly:

`return serviceExportToServiceImport(svcExport, numRequeues, syncer.Update)`

Reply:
When call `return serviceExportToServiceImport(svcExport, numRequeues, syncer.Update)` directly

[serviceExportToServiceImport code](https://github.com/submariner-io/lighthouse/blob/v0.14.0-m1/pkg/agent/controller/agent.go#L263)

```go
	if op == syncer.Update && getLastExportConditionReason(svcExport) != serviceUnavailable {
		return nil, false
	}
```

LastExportConditionReason "" may be trigger by `onSuccessfulServiceImportSync` or `serviceToRemoteServiceImport` directly call at least. will ConditionReason "" may be somthing else? If it's not,we can change onSuccessfulServiceImportSync reason with `ServiceImportSynced`.

If we do not want to change `onSuccessfulServiceImportSync` reason, change `ServiceExport` then directly call `serviceToRemoteServiceImport` will cause `serviceToRemoteServiceImport` triggered more than it's needs.

Reply:
you mean [serviceExportToServiceImport](https://github.com/submariner-io/lighthouse/blob/v0.14.0-m1/pkg/agent/controller/agent.go#L263) like this ?：

```go
	if op == syncer.Update && getLastExportCondition(svcExport) != corev1.ConditionTrue {
		return nil, false
	}
```

onSuccessfulServiceImportSync will trigger this ConditionTrue too. how distinguish service update direct call and onSuccessfulServiceImportSync? service update direct call and onSuccessfulServiceImportSync has the same Status and Reason

Reply:
https://github.com/submariner-io/lighthouse/pull/899 should alleviate your issues with calling `serviceToRemoteServiceImport` directly.

Reply:
OK, i will have a look